fe/fe-core/src/main/java/com/starrocks/load/streamload/StreamLoadMgr.java

```diff
@@ -228,11 +229,27 @@ public Database checkDbName(String dbName) throws UserException {
         return db;
     }
 
-    // add load tasks and also add to to callback factory
+    // add load tasks and also add callback factory
     public void addLoadTask(StreamLoadTask task) {
         if (task.isSyncStreamLoad()) {
             txnIdToSyncStreamLoadTasks.put(task.getTxnId(), task);
         }
+
+        // Clear the stream load tasks manually
+        if (idToStreamLoadTask.size() > Config.label_keep_max_num) {
+            // If enable_load_profile = true,
+            // most stream load tasks are generated through flink-cdc and routine load generally,
+            // so clearing the syncStreamLoadTask is preferred.
+            LOG.info("trigger cleanSyncStreamLoadTasks when add load task label:{}",task.getLabel());
+            cleanSyncStreamLoadTasks();
+            // The size of idToStreamLoadTask is still huge, indicates that the type of most tasks is PARALLEL,
+            // so clean all the streamLoadTasks manaully not waitting for Config.label_keep_max_second.
+            if (idToStreamLoadTask.size() > Config.label_keep_max_num / 2) {
```

Comment:
Why divide 2 in this place?

Reply:
Want to clear tasks that are not of type PARALLEL firstly, if there are still  1/2 tasks left， indicating that there are many tasks of type PARALLEL, so clear all of them again。Otherwise，clean the task of not type of PARALLEL is enough

Reply:
@smartlxh  In previous implementation, `cleanOldStreamLoadTasks()` is called periodically with 4 hour as default scheduling time period. In this pr, after `Config.label_keep_max_num` check, `cleanOldStreamLoadTasks()` will be called immediately while condition is matched. However task is not cleaned right away, it still need to wait `Config.label_keep_max_second` seconds before it is cleaned. This is the expected behavior, right?

Reply:
Ok, I have changed the interface to support the clean without waitting